### PR TITLE
Issue #5044 - Refactor of Jetty WebSocket Upgrade Request/Response

### LINKED ISSUE #5044: Jetty WebSocket UpgradeRequest & UpgradeResponse types in Jetty 10
**Jetty version**
10.0.x

**Description**
The `UpgradeRequest` and `UpgradeResponse` types in jetty 10 seem like they could use some simplification.

There is the API `UpgradeRequest` and `UpgradeResponse`. The Server API `JettyServerUpgradeRequest` and `JettyServerUpgradeResponse` which do not extend the API interfaces. Then there are heaps of Adapters and Delegate classes for these to convert between different interfaces. And some of the methods on these remain unimplemented.

### PR DESCRIPTION
**Pull request by lachlan-roberts**
**Issue #5044**

The Jetty WebSocket API Interfaces `UpgradeRequest`, `UpgradeResponse` have been simplified with some server specific methods being moved to `JettyServerUpgradeRequest` and `JettyServerUpgradeResponse` which are now interfaces which extend the API interface.

This allowed me to remove all the empty and `UnsupportedOperationException` implementations in the various implementations.

### CODE REVIEW

File: jetty-websocket/websocket-jetty-api/src/main/java/org/eclipse/jetty/websocket/api/UpgradeRequest.java [hunk 1]
```diff
@@ -31,26 +31,6 @@
...
-    /**
-     * Add WebSocket Extension Configuration(s) to request
-     * <p>
-     * This is merely the list of requested Extensions to use, see {@link UpgradeResponse#getExtensions()} for what was
-     * negotiated
-     *
-     * @param configs the configuration(s) to add
-     */
-    void addExtensions(String... configs);
-
```
Comment (joakime) [inline jetty-websocket/websocket-jetty-api/src/main/java/org/eclipse/jetty/websocket/api/UpgradeRequest.java:53]: The changes in this file specifically bother me, but I cannot articulate why (yet).

File: jetty-websocket/websocket-jetty-api/src/main/java/org/eclipse/jetty/websocket/api/UpgradeResponse.java [hunk 2]
```diff
@@ -88,57 +79,4 @@
...
-     * @param value the header value
-     */
-    void setHeader(String name, String value);
-
-    /**
-     * Set the HTTP Response status code
-     *
-     * @param statusCode the status code
-     */
-    void setStatusCode(int statusCode);
```
Comment (joakime) [inline jetty-websocket/websocket-jetty-api/src/main/java/org/eclipse/jetty/websocket/api/UpgradeResponse.java:143]: Yeah ... this api interface being this useless is also a point of concern for me.

Comment (lachlan-roberts) [reply jetty-websocket/websocket-jetty-api/src/main/java/org/eclipse/jetty/websocket/api/UpgradeResponse.java:143]: Why is it now useless? it still has all the getters.

I could combine this interface with the Server extension of it, but all these methods on the client side would just be implemented with `UnsupportedOperationException`.

The same issue goes for the setters on the `ClientUpgradeRequest` which are not in the `UpgradeRequest` interface, as they would be unimplementable on the server side.

**Review by joakime — APPROVED** [review]
After a hangout review with @lachlan-roberts i'm good with this change as it stands.

+1 from me.

**Pull request merged**